File: design-patterns/go/composite/player_group/player_group.go
// player_group.go

package main

type PlayerGroup struct {
	playerList []Player
}

func NewPlayerGroup() (playerGroup *PlayerGroup) {
	playerGroup = &PlayerGroup{}
	return
}

func (playerGroup *PlayerGroup) AddElement(player Player) {
	playerGroup.playerList = append(playerGroup.playerList, player)
}

func (playerGroup *PlayerGroup) RemoveElement(player Player) {
	newPlayerList := []Player{}

	for _, currentPlayer := range playerGroup.playerList {
		if currentPlayer == player {
			continue
		}

		newPlayerList = append(newPlayerList, currentPlayer)
	}

	playerGroup.playerList = newPlayerList
}

func (playerGroup *PlayerGroup) PrintDetails() {
	for _, player := range playerGroup.playerList {
		player.PrintDetails()
	}
}

File: design-patterns/php/composite/playergroup/PlayerGroup.php
<?php
// PlayerGroup.php

namespace BigBoxCode\DesignPattern\Composite\PlayerGroup;

class PlayerGroup implements Player {
    private array $playerList = [];

    public function printDetails(): void {
        foreach ($this->playerList as $player) {
            $player->printDetails();
        }
    }

    public function addElement(Player $player): void {
        array_push($this->playerList, $player);
    }

    public function removeElement(Player $player): void {
        $index = array_search($player, $this->playerList);

        if ($index !== false) {
            unset($this->playerList[$index]);
        }
    }
}